crates/holochain/src/conductor/interface/websocket.rs

```diff
@@ -249,24 +342,139 @@ pub mod test {
+        let _rx = PollRecv::new::<AdminResponse>(rx);
+
+        let agent_key = conductor_handle
+            .keystore()
+            .new_sign_keypair_random()
+            .await
+            .unwrap();
+
+        let (dna_file, _, _) =
+            SweetDnaFile::unique_from_test_wasms(vec![TestWasm::EmitSignal]).await;
+        let app_bundle = app_bundle_from_dnas([&dna_file]).await;
+        let request = AdminRequest::InstallApp(Box::new(InstallAppPayload {
+            source: AppBundleSource::Bundle(app_bundle),
+            agent_key: agent_key.clone(),
+            installed_app_id: None,
+            membrane_proofs: HashMap::new(),
+            network_seed: None,
+            ignore_genesis_failure: false,
+        }));
+        let response: AdminResponse = admin_tx.request(request).await.unwrap();
+        let app_info = match response {
+            AdminResponse::AppInstalled(app_info) => app_info,
+            _ => panic!("didn't install app"),
+        };
+        let cell_id = match &app_info
+            .cell_info
+            .get(&dna_file.dna_hash().to_string())
+            .unwrap()[0]
+        {
+            CellInfo::Provisioned(cell) => cell.cell_id.clone(),
+            _ => panic!("emit_signal cell not available"),
+        };
+
+        // Activate cells
+        let request = AdminRequest::EnableApp {
+            installed_app_id: app_info.installed_app_id.clone(),
+        };
+        let response: AdminResponse = admin_tx.request(request).await.unwrap();
+        assert_matches!(response, AdminResponse::AppEnabled { .. });
+
+        // Attach App Interface
+        let request = AdminRequest::AttachAppInterface { port: None };
+        let response: AdminResponse = admin_tx.request(request).await.unwrap();
+        let app_port = match response {
+            AdminResponse::AppInterfaceAttached { port } => port,
+            _ => panic!("app interface couldn't be attached"),
+        };
+
+        let (app_tx, rx) = websocket_client_by_port(app_port).await.unwrap();
+        let _rx = PollRecv::new::<AppResponse>(rx);
+
+        // Call Zome
+        let (nonce, expires_at) = holochain_nonce::fresh_nonce(Timestamp::now()).unwrap();
+        let request = AppRequest::CallZome(Box::new(
+            ZomeCall::try_from_unsigned_zome_call(
+                conductor_handle.keystore(),
+                ZomeCallUnsigned {
+                    provenance: agent_key.clone(),
+                    cell_id: cell_id.clone(),
+                    zome_name: TestWasm::EmitSignal.coordinator_zome_name(),
+                    fn_name: "commit_entry_and_emit_signal_post_commit".into(),
+                    cap_secret: None,
+                    payload: ExternIO::encode(()).unwrap(),
+                    nonce,
+                    expires_at,
+                },
+            )
+            .await
+            .unwrap(),
+        ));
+        let _: AppResponse = app_tx.request(request).await.unwrap();
```

Comment:
Can you check that the signal was received?

Reply:
done